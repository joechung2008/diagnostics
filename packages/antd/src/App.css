html,
body {
  height: 100%;
  margin: 0;
  padding: 0;
}

*,
*::after,
*::before {
  box-sizing: border-box;
}

.centered {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;
}

.extension-root {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  max-height: max-content;
  overflow-y: auto;
}

.flexbox {
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.grow {
  flex: 1;
}

.stack {
  display: flex;
  flex-direction: row;
  height: 100%;
}

.tab-list {
  padding-left: 1rem;
}

.tab-panel {
  flex: 1;
  overflow-y: auto;
}
