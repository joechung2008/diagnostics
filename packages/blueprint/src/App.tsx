import {
  Button,
  ButtonGroup,
  NonIdealState,
  Spinner,
  Tab,
  Tabs,
} from "@blueprintjs/core";
import clsx from "clsx";
import { useCallback, useMemo, useState } from "react";
import useSWR from "swr";
import BuildInfo from "./BuildInfo";
import Extension from "./Extension";
import Extensions from "./Extensions";
import ServerInfo from "./ServerInfo";
import { isExtensionInfo } from "./utils";

const Environment = {
  Public: "https://hosting.portal.azure.net/api/diagnostics",
  Fairfax: "https://hosting.azureportal.usgovcloudapi.net/api/diagnostics",
  Mooncake: "https://hosting.azureportal.chinacloudapi.cn/api/diagnostics",
} as const;

type EnvironmentUrl =
  | "https://hosting.portal.azure.net/api/diagnostics"
  | "https://hosting.azureportal.usgovcloudapi.net/api/diagnostics"
  | "https://hosting.azureportal.chinacloudapi.cn/api/diagnostics";

interface AppProps {
  className?: string;
}

const App = ({ className }: AppProps) => {
  const [environment, setEnvironment] = useState<EnvironmentUrl>(
    Environment.Public
  );
  const [extension, setExtension] = useState<ExtensionInfo>();
  const [selectedTab, setSelectedTab] = useState<string>("extensions");

  const {
    data: diagnostics,
    error,
    isLoading,
  } = useSWR(environment, async (environment: string) => {
    const response = await fetch(environment);
    if (!response.ok) {
      throw new Error(`Failed to fetch diagnostics: ${response.statusText}`);
    }

    return response.json();
  });

  const showPaasServerless = useMemo(
    () => isExtensionInfo(diagnostics?.extensions["paasserverless"]),
    [diagnostics?.extensions]
  );

  const handleLinkClick = useCallback(
    (item?: KeyedNavLink) => {
      if (item) {
        const extension = diagnostics?.extensions[item.key];
        if (isExtensionInfo(extension)) {
          setExtension(extension);
        }
      }
    },
    [diagnostics?.extensions]
  );

  return (
    <div className={clsx("flexbox", className)}>
      <ButtonGroup>
        <Button
          active={environment === Environment.Public}
          variant="minimal"
          onClick={() => {
            setEnvironment(Environment.Public);
            setExtension(undefined);
            setSelectedTab("extensions");
          }}
        >
          Public Cloud
        </Button>
        <Button
          active={environment === Environment.Fairfax}
          variant="minimal"
          onClick={() => {
            setEnvironment(Environment.Fairfax);
            setExtension(undefined);
            setSelectedTab("extensions");
          }}
        >
          Fairfax
        </Button>
        <Button
          active={environment === Environment.Mooncake}
          variant="minimal"
          onClick={() => {
            setEnvironment(Environment.Mooncake);
            setExtension(undefined);
            setSelectedTab("extensions");
          }}
        >
          Mooncake
        </Button>
        <span className="button-group-separator" />
        {showPaasServerless && (
          <Button
            variant="minimal"
            onClick={() => {
              const paasserverless = diagnostics?.extensions["paasserverless"];
              if (isExtensionInfo(paasserverless)) {
                setExtension(paasserverless);
                setSelectedTab("extensions");
              }
            }}
          >
            paasserverless
          </Button>
        )}
        <Button
          variant="minimal"
          onClick={() => {
            const websites = diagnostics?.extensions["websites"];
            if (isExtensionInfo(websites)) {
              setExtension(websites);
              setSelectedTab("extensions");
            }
          }}
        >
          websites
        </Button>
      </ButtonGroup>
      <Tabs
        selectedTabId={selectedTab}
        size="large"
        onChange={(tabId) => setSelectedTab(tabId as string)}
      >
        <Tab
          id="extensions"
          aria-controls="extensions-tab"
          title="Extensions"
        />
        <Tab id="build" aria-controls="build-tab" title="Build Information" />
        <Tab
          id="server"
          aria-controls="server-tab"
          title="Server Information"
        />
      </Tabs>
      {isLoading ? (
        <div className="tab-panel vertical-center">
          <Spinner aria-label="Loading..." />
        </div>
      ) : error ? (
        <NonIdealState icon="error" title="Error" description={error.message} />
      ) : selectedTab === "extensions" && diagnostics?.extensions ? (
        <div id="extensions-tab" className="tab-panel">
          <div className="stack">
            <Extensions
              extensions={diagnostics.extensions}
              onLinkClick={handleLinkClick}
            />
            {extension && <Extension {...extension} />}
          </div>
        </div>
      ) : selectedTab === "build" && diagnostics?.buildInfo ? (
        <div id="build-tab" className="tab-panel">
          <BuildInfo {...diagnostics.buildInfo} />
        </div>
      ) : selectedTab === "server" && diagnostics?.serverInfo ? (
        <div id="server-tab" className="tab-panel">
          <ServerInfo {...diagnostics.serverInfo} />
        </div>
      ) : null}
    </div>
  );
};

export default App;
