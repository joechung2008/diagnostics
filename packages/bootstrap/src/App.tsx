import { Activity, useCallback, useMemo, useState } from "react";
import useSWR from "swr";
import {
  Alert,
  Button,
  ButtonGroup,
  Dropdown,
  Nav,
  Spinner,
} from "react-bootstrap";
import BuildInfo from "./BuildInfo";
import Extension from "./Extension";
import Extensions from "./Extensions";
import ServerInfo from "./ServerInfo";
import { isExtensionInfo } from "./utils";
import { useSystemTheme } from "./useSystemTheme";

const Environment = {
  Public: "https://hosting.portal.azure.net/api/diagnostics",
  Fairfax: "https://hosting.azureportal.usgovcloudapi.net/api/diagnostics",
  Mooncake: "https://hosting.azureportal.chinacloudapi.cn/api/diagnostics",
} as const;

type EnvironmentUrl = (typeof Environment)[keyof typeof Environment];

const App: React.FC = () => {
  const [extension, setExtension] = useState<ExtensionInfo>();
  const [environment, setEnvironment] = useState<EnvironmentUrl>(
    Environment.Public
  );
  const [selectedTab, setSelectedTab] = useState<string>("extensions");

  // Use system theme hook to set data-bs-theme on html element
  useSystemTheme();

  const {
    data: diagnostics,
    error,
    isLoading,
  } = useSWR<Diagnostics>(environment, async (environment: string) => {
    const response = await fetch(environment);
    if (!response.ok) {
      throw new Error(`Error fetching diagnostics: ${response.statusText}`);
    }

    return response.json();
  });

  const environmentName = useMemo(() => {
    switch (environment) {
      case Environment.Public:
        return "Public Cloud";
      case Environment.Fairfax:
        return "Fairfax";
      case Environment.Mooncake:
        return "Mooncake";
      default:
        return "Select environment";
    }
  }, [environment]);

  const environments = useMemo(
    () => [
      {
        key: "public",
        text: "Public Cloud",
        selected: environment === Environment.Public,
        onClick: () => {
          setEnvironment(Environment.Public);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "fairfax",
        text: "Fairfax",
        selected: environment === Environment.Fairfax,
        onClick: () => {
          setEnvironment(Environment.Fairfax);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "mooncake",
        text: "Mooncake",
        selected: environment === Environment.Mooncake,
        onClick: () => {
          setEnvironment(Environment.Mooncake);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
    ],
    [environment]
  );

  const showPaasServerless = useMemo(
    () => isExtensionInfo(diagnostics?.extensions?.["paasserverless"]),
    [diagnostics]
  );

  const handleLinkClick = useCallback(
    (_?: React.MouseEvent, item?: KeyedNavLink) => {
      if (item) {
        const extension = diagnostics?.extensions[item.key];
        if (isExtensionInfo(extension)) {
          setExtension(extension);
        }
      }
    },
    [diagnostics?.extensions]
  );

  return (
    <div className="flexbox">
      <ButtonGroup className="w-auto align-self-start">
        <Dropdown>
          <Dropdown.Toggle variant="outline-primary" id="environment-dropdown">
            {environmentName}
          </Dropdown.Toggle>
          <Dropdown.Menu>
            {environments.map((env) => (
              <Dropdown.Item
                key={env.key}
                active={env.selected}
                onClick={env.onClick}
              >
                {env.text}
              </Dropdown.Item>
            ))}
          </Dropdown.Menu>
        </Dropdown>
        {showPaasServerless && (
          <Button
            variant="outline-secondary"
            onClick={() => {
              const paasserverless = diagnostics?.extensions["paasserverless"];
              if (isExtensionInfo(paasserverless)) {
                setExtension(paasserverless);
                setSelectedTab("extensions");
              }
            }}
          >
            paasserverless
          </Button>
        )}
        <Button
          variant="outline-secondary"
          onClick={() => {
            const websites = diagnostics?.extensions["websites"];
            if (isExtensionInfo(websites)) {
              setExtension(websites);
              setSelectedTab("extensions");
            }
          }}
        >
          websites
        </Button>
      </ButtonGroup>
      <Nav
        variant="tabs"
        activeKey={selectedTab}
        onSelect={(selectedKey) => setSelectedTab(selectedKey || "extensions")}
      >
        <Nav.Item>
          <Nav.Link eventKey="extensions">Extensions</Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link eventKey="build">Build Information</Nav.Link>
        </Nav.Item>
        <Nav.Item>
          <Nav.Link eventKey="server">Server Information</Nav.Link>
        </Nav.Item>
      </Nav>
      <Activity mode={isLoading && !error ? "visible" : "hidden"}>
        <div className="tab-panel centered">
          <Spinner animation="border" />
        </div>
      </Activity>
      <Activity mode={error ? "visible" : "hidden"}>
        <div className="tab-panel centered">
          <Alert variant="danger">
            An error occurred while loading diagnostics.
          </Alert>
        </div>
      </Activity>
      <Activity
        mode={
          selectedTab === "extensions" && !error && !isLoading
            ? "visible"
            : "hidden"
        }
      >
        {diagnostics?.extensions && (
          <div className="tab-panel">
            <div className="stack">
              <Extensions
                extensions={diagnostics.extensions}
                onLinkClick={handleLinkClick}
              />
              {extension && <Extension {...extension} />}
            </div>
          </div>
        )}
      </Activity>
      <Activity
        mode={
          selectedTab === "build" && !error && !isLoading ? "visible" : "hidden"
        }
      >
        {diagnostics?.buildInfo && (
          <div className="tab-panel">
            <BuildInfo {...diagnostics.buildInfo} />
          </div>
        )}
      </Activity>
      <Activity
        mode={
          selectedTab === "server" && !error && !isLoading
            ? "visible"
            : "hidden"
        }
      >
        {diagnostics?.serverInfo && (
          <div className="tab-panel">
            <ServerInfo {...diagnostics!.serverInfo} />
          </div>
        )}
      </Activity>
    </div>
  );
};

export default App;
