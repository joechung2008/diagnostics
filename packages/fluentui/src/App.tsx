import {
  Menu,
  MenuButton,
  MenuItemRadio,
  MenuList,
  MenuPopover,
  MenuTrigger,
  Tab,
  TabList,
  Toolbar,
  ToolbarButton,
  type MenuCheckedValueChangeData,
  type MenuCheckedValueChangeEvent,
} from "@fluentui/react-components";
import { startTransition, useCallback, useMemo, useState } from "react";
import useSWR from "swr";
import BuildInfo from "./BuildInfo";
import Extension from "./Extension";
import Extensions from "./Extensions";
import ServerInfo from "./ServerInfo";
import { isExtensionInfo } from "./utils";

type Environment = (typeof Environment)[keyof typeof Environment];

const Environment = {
  Public: "https://hosting.portal.azure.net/api/diagnostics",
  Fairfax: "https://hosting.azureportal.usgovcloudapi.net/api/diagnostics",
  Mooncake: "https://hosting.azureportal.chinacloudapi.cn/api/diagnostics",
} as const;

function getEnvironnmentName(environment: Environment | undefined): string {
  switch (environment) {
    case Environment.Public:
      return "Public Cloud";
    case Environment.Fairfax:
      return "Fairfax";
    case Environment.Mooncake:
      return "Mooncake";
    default:
      return "Select environment";
  }
}

const App: React.FC = () => {
  const [environments, setEnvironments] = useState<
    Record<string, Environment[]>
  >({
    environment: [Environment.Public],
  });
  const environment = useMemo<Environment>(
    () => environments.environment[0],
    [environments.environment]
  );
  const {
    data: diagnostics,
    error,
    isLoading,
  } = useSWR(
    environment,
    async (url: string) => {
      const response = await fetch(url);
      if (!response.ok) {
        throw new Error(`Failed to fetch diagnostics: ${response.statusText}`);
      }
      return response.json();
    },
    {
      suspense: true,
    }
  );

  const [extension, setExtension] = useState<ExtensionInfo>();
  const [selectedTab, setSelectedTab] = useState<string>("extensions");

  const handleLinkClick: ExtensionsProps["onLinkClick"] = useCallback(
    (_, item) => {
      if (item) {
        const $extension = diagnostics?.extensions[item.key];
        if (isExtensionInfo($extension)) {
          setExtension($extension);
        }
      }
    },
    [diagnostics?.extensions]
  );

  const showPaasServerless = useMemo(
    () => isExtensionInfo(diagnostics?.extensions["paasserverless"]),
    [diagnostics?.extensions]
  );

  const environmentName = useMemo(
    () => getEnvironnmentName(environment),
    [environment]
  );

  const handleEnvironmentChange = useCallback(
    (_: MenuCheckedValueChangeEvent, data: MenuCheckedValueChangeData) => {
      if (data.checkedItems?.length ?? 0 > 0) {
        startTransition(() => {
          setEnvironments((previous) => ({
            ...previous,
            [data.name]: data.checkedItems as Environment[],
          }));
          setExtension(undefined);
        });
      }
    },
    []
  );

  if (isLoading) {
    return <div>Loading diagnostics...</div>;
  }

  if (error) {
    return <div>Error loading diagnostics: {error.message}</div>;
  }

  if (!diagnostics) {
    return null;
  }

  return (
    <div className="flexbox">
      <Toolbar>
        <Menu
          checkedValues={environments}
          onCheckedValueChange={handleEnvironmentChange}
        >
          <MenuTrigger disableButtonEnhancement>
            <MenuButton>{environmentName}</MenuButton>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              {Object.entries(Environment).map(([key, value]) => (
                <MenuItemRadio key={key} name="environment" value={value}>
                  {getEnvironnmentName(value)}
                </MenuItemRadio>
              ))}
            </MenuList>
          </MenuPopover>
        </Menu>
        {showPaasServerless && (
          <ToolbarButton
            key="paasserverless"
            onClick={() => {
              const paasserverless = diagnostics?.extensions["paasserverless"];
              if (isExtensionInfo(paasserverless)) {
                setExtension(paasserverless);
              }
            }}
          >
            paasserverless
          </ToolbarButton>
        )}
        <ToolbarButton
          key="websites"
          onClick={() => {
            const websites = diagnostics?.extensions["websites"];
            if (isExtensionInfo(websites)) {
              setExtension(websites);
            }
          }}
        >
          websites
        </ToolbarButton>
      </Toolbar>
      <TabList
        selectedValue={selectedTab}
        onTabSelect={(_, data) => setSelectedTab(data.value as string)}
      >
        <Tab value="extensions">Extensions</Tab>
        <Tab value="build">Build Information</Tab>
        <Tab value="server">Server Information</Tab>
      </TabList>
      {selectedTab === "extensions" && diagnostics?.extensions && (
        <div className="tab-panel">
          <div className="stack">
            <Extensions
              extensions={diagnostics.extensions}
              onLinkClick={handleLinkClick}
            />
            {extension && <Extension {...extension} />}
          </div>
        </div>
      )}
      {selectedTab === "build" && diagnostics?.buildInfo && (
        <div className="tab-panel">
          <BuildInfo {...diagnostics.buildInfo} />
        </div>
      )}
      {selectedTab === "server" && diagnostics?.serverInfo && (
        <div className="tab-panel">
          <ServerInfo {...diagnostics.serverInfo} />
        </div>
      )}
    </div>
  );
};

export default App;
