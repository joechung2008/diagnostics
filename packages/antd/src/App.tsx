import { Empty, Flex, Menu, Spin, Tabs } from "antd";
import { Activity, useCallback, useMemo, useState } from "react";
import useSWR from "swr";
import BuildInfo from "./BuildInfo";
import Extension from "./Extension";
import Extensions from "./Extensions";
import ServerInfo from "./ServerInfo";
import { isExtensionInfo, when } from "./utils";

import "./App.css";

const Environment = {
  Public: "https://hosting.portal.azure.net/api/diagnostics",
  Fairfax: "https://hosting.azureportal.usgovcloudapi.net/api/diagnostics",
  Mooncake: "https://hosting.azureportal.chinacloudapi.cn/api/diagnostics",
} as const;

type EnvironmentUrl = (typeof Environment)[keyof typeof Environment];

const items = [
  { key: "extensions", label: "Extensions" },
  { key: "build", label: "Build Information" },
  { key: "server", label: "Server Information" },
];

const App: React.FC = () => {
  const [extension, setExtension] = useState<ExtensionInfo>();
  const [environment, setEnvironment] = useState<EnvironmentUrl>(
    Environment.Public
  );
  const [selectedTab, setSelectedTab] = useState("extensions");

  const {
    data: diagnostics,
    error,
    isLoading,
  } = useSWR<Diagnostics>(environment, async (environment: string) => {
    const response = await fetch(environment);

    if (!response.ok) {
      throw new Error(`Error fetching diagnostics: ${response.statusText}`);
    }

    return response.json();
  });

  const environmentName = useMemo(() => {
    switch (environment) {
      case Environment.Public:
        return "Public Cloud";
      case Environment.Fairfax:
        return "Fairfax";
      case Environment.Mooncake:
        return "Mooncake";
      default:
        return "Select environment";
    }
  }, [environment]);

  const showPaasServerless = useMemo(
    () => isExtensionInfo(diagnostics?.extensions["paasserverless"]),
    [diagnostics?.extensions]
  );

  const environments = useMemo(
    () => [
      {
        key: "public",
        text: "Public Cloud",
        selected: environment === Environment.Public,
        onClick: () => {
          setEnvironment(Environment.Public);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "fairfax",
        text: "Fairfax",
        selected: environment === Environment.Fairfax,
        onClick: () => {
          setEnvironment(Environment.Fairfax);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "mooncake",
        text: "Mooncake",
        selected: environment === Environment.Mooncake,
        onClick: () => {
          setEnvironment(Environment.Mooncake);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
    ],
    [environment]
  );

  const menuItems = useMemo(
    () => [
      {
        key: "env-dropdown",
        label: environmentName,
        children: environments.map((env) => ({
          key: env.key,
          label: env.text,
          onClick: env.onClick,
        })),
      },
      ...when(showPaasServerless, {
        key: "paasserverless",
        label: "paasserverless",
        onClick: () => {
          const paasserverless = diagnostics?.extensions["paasserverless"];
          if (isExtensionInfo(paasserverless)) {
            setExtension(paasserverless);
            setSelectedTab("extensions");
          }
        },
      }),
      {
        key: "websites",
        label: "websites",
        onClick: () => {
          const websites = diagnostics?.extensions["websites"];
          if (isExtensionInfo(websites)) {
            setExtension(websites);
            setSelectedTab("extensions");
          }
        },
      },
    ],
    [environments, environmentName, showPaasServerless, diagnostics?.extensions]
  );

  const handleLinkClick = useCallback(
    (_?: React.MouseEvent, item?: KeyedNavLink) => {
      if (item) {
        const extension = diagnostics?.extensions[item.key];
        if (isExtensionInfo(extension)) {
          setExtension(extension);
        }
      }
    },
    [diagnostics?.extensions]
  );

  return (
    <div className="flexbox">
      <Menu mode="horizontal" items={menuItems} />
      <Tabs
        activeKey={selectedTab}
        className="tab-list"
        items={items}
        onChange={(key) => setSelectedTab(key)}
      />
      <Activity mode={isLoading ? "visible" : "hidden"}>
        <div className="tab-panel">
          <Flex className="centered">
            <Spin size="large" />
          </Flex>
        </div>
      </Activity>
      <Activity mode={error && !isLoading ? "visible" : "hidden"}>
        {error && (
          <div className="tab-panel">
            <Flex className="centered">
              <Empty
                description={`Error: ${error.message ?? "Unexpected error"}`}
              />
            </Flex>
          </div>
        )}
      </Activity>
      <Activity
        mode={
          !error && !isLoading && selectedTab === "extensions"
            ? "visible"
            : "hidden"
        }
      >
        {diagnostics?.extensions && (
          <div className="tab-panel">
            <Flex className="stack">
              <Extensions
                extensions={diagnostics.extensions}
                onLinkClick={handleLinkClick}
              />
              {extension && <Extension {...extension} />}
            </Flex>
          </div>
        )}
      </Activity>
      <Activity
        mode={
          !error && !isLoading && selectedTab === "build" ? "visible" : "hidden"
        }
      >
        {diagnostics?.buildInfo && (
          <div className="tab-panel">
            <BuildInfo {...diagnostics.buildInfo} />
          </div>
        )}
      </Activity>
      <Activity
        mode={
          !error && !isLoading && selectedTab === "server"
            ? "visible"
            : "hidden"
        }
      >
        {diagnostics?.buildInfo && (
          <div className="tab-panel">
            <ServerInfo {...diagnostics.serverInfo} />
          </div>
        )}
      </Activity>
    </div>
  );
};

export default App;
