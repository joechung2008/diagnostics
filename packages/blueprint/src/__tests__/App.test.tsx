import { render } from "@testing-library/react";
import useSWR from "swr";
import { afterEach, describe, expect, it, vi } from "vitest";
import App from "../App";

vi.mock("swr");

describe("App", () => {
  const mockUseSWR = vi.mocked(useSWR);

  afterEach(() => {
    vi.clearAllMocks();
  });

  it("should render the loading state while loading", async () => {
    mockUseSWR.mockReturnValue({
      data: undefined,
      error: undefined,
      isLoading: true,
      mutate: vi.fn(),
      isValidating: false,
    });

    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the error state on fetch failure", async () => {
    mockUseSWR.mockReturnValue({
      data: undefined,
      error: new Error("Failed to fetch diagnostics"),
      isLoading: false,
      mutate: vi.fn(),
      isValidating: false,
    });

    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the initial loaded state", async () => {
    mockUseSWR.mockReturnValue({
      data: {
        buildInfo: {
          buildVersion: "1.0.0",
        },
        extensions: {
          websites: { extensionName: "websites" },
          paasserverless: { extensionName: "paasserverless" },
        },
        serverInfo: {
          deploymentId: "deployment-123",
          extensionSync: {
            totalSyncAllCount: 0,
          },
          hostname: "localhost",
          nodeVersions: "v18.0.0",
          serverId: "server-abc",
          uptime: 123456,
        },
      },
      error: undefined,
      isLoading: false,
      mutate: vi.fn(),
      isValidating: false,
    });

    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
