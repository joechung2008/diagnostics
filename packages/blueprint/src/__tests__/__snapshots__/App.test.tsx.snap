// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render the error state on fetch failure 1`] = `
<DocumentFragment>
  <div
    class="flexbox"
  >
    <div
      class="bp6-button-group"
    >
      <button
        class="bp6-button bp6-active bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          Public Cloud
        </span>
      </button>
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          Fairfax
        </span>
      </button>
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          Mooncake
        </span>
      </button>
      <span
        class="button-group-separator"
      />
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          websites
        </span>
      </button>
    </div>
    <div
      class="bp6-tabs"
    >
      <div
        class="bp6-tab-list bp6-large"
        role="tablist"
      >
        <div
          class="bp6-tab-indicator-wrapper"
          style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px; transition: none;"
        >
          <div
            class="bp6-tab-indicator"
          />
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_extensions"
          aria-disabled="false"
          aria-expanded="true"
          aria-selected="true"
          class="bp6-tab"
          data-tab-id="extensions"
          id="bp6-tab-title_undefined_extensions"
          role="tab"
          tabindex="0"
        >
          Extensions
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_build"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp6-tab"
          data-tab-id="build"
          id="bp6-tab-title_undefined_build"
          role="tab"
          tabindex="-1"
        >
          Build Information
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_server"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp6-tab"
          data-tab-id="server"
          id="bp6-tab-title_undefined_server"
          role="tab"
          tabindex="-1"
        >
          Server Information
        </div>
      </div>
    </div>
    <div
      class="bp6-non-ideal-state bp6-non-ideal-state-vertical"
    >
      <div
        class="bp6-non-ideal-state-visual"
        style="font-size: 48px; line-height: 48px;"
      >
        <span
          aria-hidden="true"
          class="bp6-icon bp6-icon-error bp6-icon-muted"
          data-icon="error"
          tabindex="-1"
        />
      </div>
      <div
        class="bp6-non-ideal-state-text"
      >
        <h4
          class="bp6-heading"
        >
          Error
        </h4>
        <div>
          Failed to fetch diagnostics
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App > should render the initial loaded state 1`] = `
<DocumentFragment>
  <div
    class="flexbox"
  >
    <div
      class="bp6-button-group"
    >
      <button
        class="bp6-button bp6-active bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          Public Cloud
        </span>
      </button>
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          Fairfax
        </span>
      </button>
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          Mooncake
        </span>
      </button>
      <span
        class="button-group-separator"
      />
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          paasserverless
        </span>
      </button>
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          websites
        </span>
      </button>
    </div>
    <div
      class="bp6-tabs"
    >
      <div
        class="bp6-tab-list bp6-large"
        role="tablist"
      >
        <div
          class="bp6-tab-indicator-wrapper"
          style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px; transition: none;"
        >
          <div
            class="bp6-tab-indicator"
          />
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_extensions"
          aria-disabled="false"
          aria-expanded="true"
          aria-selected="true"
          class="bp6-tab"
          data-tab-id="extensions"
          id="bp6-tab-title_undefined_extensions"
          role="tab"
          tabindex="0"
        >
          Extensions
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_build"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp6-tab"
          data-tab-id="build"
          id="bp6-tab-title_undefined_build"
          role="tab"
          tabindex="-1"
        >
          Build Information
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_server"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp6-tab"
          data-tab-id="server"
          id="bp6-tab-title_undefined_server"
          role="tab"
          tabindex="-1"
        >
          Server Information
        </div>
      </div>
    </div>
    <div
      class="tab-panel"
      id="extensions-tab"
    >
      <div
        class="stack"
      >
        <div
          class="extension-root"
        >
          <div
            class="bp6-tabs bp6-vertical"
          >
            <div
              class="bp6-tab-list"
              role="tablist"
            >
              <div
                class="bp6-tab-indicator-wrapper"
                style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px; transition: none;"
              >
                <div
                  class="bp6-tab-indicator"
                />
              </div>
              <div
                aria-controls="bp6-tab-panel_undefined_paasserverless"
                aria-disabled="false"
                aria-expanded="true"
                aria-selected="true"
                class="bp6-tab"
                data-tab-id="paasserverless"
                id="bp6-tab-title_undefined_paasserverless"
                role="tab"
                tabindex="0"
              >
                paasserverless
              </div>
              <div
                aria-controls="bp6-tab-panel_undefined_websites"
                aria-disabled="false"
                aria-expanded="false"
                aria-selected="false"
                class="bp6-tab"
                data-tab-id="websites"
                id="bp6-tab-title_undefined_websites"
                role="tab"
                tabindex="-1"
              >
                websites
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App > should render the loading state while loading 1`] = `
<DocumentFragment>
  <div
    class="flexbox"
  >
    <div
      class="bp6-button-group"
    >
      <button
        class="bp6-button bp6-active bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          Public Cloud
        </span>
      </button>
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          Fairfax
        </span>
      </button>
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          Mooncake
        </span>
      </button>
      <span
        class="button-group-separator"
      />
      <button
        class="bp6-button bp6-minimal"
        type="button"
      >
        <span
          class="bp6-button-text"
        >
          websites
        </span>
      </button>
    </div>
    <div
      class="bp6-tabs"
    >
      <div
        class="bp6-tab-list bp6-large"
        role="tablist"
      >
        <div
          class="bp6-tab-indicator-wrapper"
          style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px; transition: none;"
        >
          <div
            class="bp6-tab-indicator"
          />
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_extensions"
          aria-disabled="false"
          aria-expanded="true"
          aria-selected="true"
          class="bp6-tab"
          data-tab-id="extensions"
          id="bp6-tab-title_undefined_extensions"
          role="tab"
          tabindex="0"
        >
          Extensions
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_build"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp6-tab"
          data-tab-id="build"
          id="bp6-tab-title_undefined_build"
          role="tab"
          tabindex="-1"
        >
          Build Information
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_server"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp6-tab"
          data-tab-id="server"
          id="bp6-tab-title_undefined_server"
          role="tab"
          tabindex="-1"
        >
          Server Information
        </div>
      </div>
    </div>
    <div
      class="tab-panel vertical-center"
    >
      <div
        aria-label="Loading..."
        aria-valuemax="100"
        aria-valuemin="0"
        class="bp6-spinner"
        role="progressbar"
      >
        <div
          class="bp6-spinner-animation"
        >
          <svg
            height="50"
            stroke-width="8.00"
            viewBox="1.00 1.00 98.00 98.00"
            width="50"
          >
            <path
              class="bp6-spinner-track"
              d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
            />
            <path
              class="bp6-spinner-head"
              d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
              pathLength="280"
              stroke-dasharray="280 280"
              stroke-dashoffset="210"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
